app/controllers/gems_adoptions_controller.rb

```diff
@@ -0,0 +1,21 @@
+class GemsAdoptionsController < ApplicationController
+  
+  def index
+    @gems_adoptions = GemsAdoption.all
+  end
+
+  def create
+    @gems_adoption = GemsAdoption.new
+    @gems_adoption.gem = params[:name]
+    @gems_adoption.description = params[:info]
+    @gems_adoption.user_id = current_user.id
+    @gems_adoption.save
+    redirect_to gem_path(id: params[:name])
+  end
+
+  def destroy
+    GemsAdoption.find(params[:id]).destroy
```

Comment:
Here we should ensure that the Gem that is going to be removed from adoption belongs to the user that is attempting to remove it. To ensure that, we have check for the given Gem in the gems put up for adoption by the current_user:

```ruby
def destroy
    current_user.gems_adoptions.find(params[:id]).destroy
    # ...
end
```